tests/test_core.py

```diff
@@ -2491,6 +2491,7 @@ def test_pthread_equal(self):
 
   @node_pthreads
   def test_pthread_dispatch_after_exit(self):
+    self.set_setting('PTHREAD_POOL_SIZE', 1)
```

Comment:
You mentioned this test hit a warning without this - what was the warning?

Reply:
```
Tried to spawn a new thread, but the thread pool is exhausted.
This might result in a deadlock unless some threads eventually exit or the code explicitly breaks out to the event loop.
If you want to increase the pool size, use setting `-s PTHREAD_POOL_SIZE=...`.
If you want to throw an explicit error instead of the risk of deadlocking in those cases, use setting `-s PTHREAD_POOL_SIZE_STRICT=2`.
```

I think I can/should revert this line form this PR though.. and instead add a real test of this bugfix